### PR TITLE
feat(flow-item): add component tokens

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #7180

## Summary

- Add component tokens for `calcite-flow-item`
- Add story

Adds the following component tokens:

 - `--calcite-flow-item-background-color`
 - `--calcite-flow-item-color`

### CODE REVIEW

File: packages/calcite-components/src/components/flow-item/flow-item.scss [hunk 1]
```diff
@@ -3,22 +3,36 @@
  *
  * These properties can be overridden using the component's tag as selector.
  *
+ * @prop --calcite-flow-item-background-color: Specifies the background of the component.
+ * @prop --calcite-flow-item-color: Specifies the color of the component.
```
Comment (macandcheese) [inline packages/calcite-components/src/components/flow-item/flow-item.scss:7]: Do we want `flow-item-header-color` and `flow-item-description-color` here?

**Comment by driskull** [comment]
Hey @macandcheese sorry this one's not ready for review yet. We need the panel to be done first.

**Comment by macandcheese** [comment]
Sorry @driskull clicked into the wrong PR 🙃

File: packages/calcite-components/src/components/flow-item/flow-item.scss (hunk 1 above)
Comment (alisonailea) [reply packages/calcite-components/src/components/flow-item/flow-item.scss:7]: not unless there have been requests for unique styling for those elements.

Comment (alisonailea) [inline packages/calcite-components/src/components/flow-item/flow-item.scss:7]: this should be updated to `--calcite-flow-item-text-color`

**Review by alisonailea — CHANGES_REQUESTED** [review]
🎉 
one minor naming update

**Pull request closed without merging**